assignments/lambda-classes.js

```diff
@@ -1 +1,191 @@
 // CODE here for your Lambda Classes
+class Person{
+  constructor(person_attr){
+    this.name = person_attr.name;
+    this.age = person_attr.age;
+    this.location = person_attr.location;
+    this.gender = person_attr.gender
+  }
+
+  speak() {
+    return `Hello my name is ${this.name}, I am from ${this.location}}`;
+  }
+}
+
+class Instructor extends Person{
+  constructor(inst_attr){
+    super(inst_attr);
+    this.specialty = inst_attr.specialty;
+    this.favLanguage = inst_attr.favLanguage;
+    this.catchPhrase = inst_attr.catchPhrase;
+  }
+
+  demo(subject) {
+    return `Today we are learning about ${subject}`;
+  }
+
+  grade(student, subject){
+    return `${student.name} receives a perfect score on ${subject}`;
+  }
+
+  gradeChange(student){
+    if(getRandomInt(2)) {
+      student.grade += Math.round(getRandomBetween(1, 10));
+    } else {
+      student.grade -= Math.round(getRandomBetween(1, 10));
+    }
+
+    if(student.grade > 100) { student.grade = 100 }
+    if(student.grade < 0) {student.grade = 0}
+
+    return `${student.name}'s grade is ${Math.round(student.grade)}`
+  }
+}
+
+class Student extends Person {
+  constructor(stu_attr){
+    super(stu_attr);
+    this.previousBackground = stu_attr.previousBackground;
+    this.className = stu_attr.className;
+    this.favSubjects = stu_attr.favSubjects;
+    this.grade = stu_attr.grade; // STRETCH
+  }
+
+  listsubjects() {
+    this.favSubjects.forEach( el => console.log(`${el}\n`));
```

Comment:
careful with stuff like this. When you are running a console log in a for each, each item gets it's own console log on it's own line automatically.